pkg/auth/dependency/authenticator/password/provider.go

```diff
@@ -51,20 +54,37 @@ func (p *Provider) New(userID string, password string) (*Authenticator, error) {
 			return nil, err
 		}
 
-		hash, err = pwd.Hash([]byte(password))
-		if err != nil {
-			panic(errors.Newf("password: failed to hash password: %w", err))
-		}
+		authen = p.populatePasswordHash(authen, password)
 	} else {
-		hash = nil
+		authen.PasswordHash = nil
+	}
+	return authen, nil
+}
+
+// WithPassword return new authenticator pointer if password is changed
+// Otherwise original authenticator will be returned
+func (p *Provider) WithPassword(userID string, a *Authenticator, password string) (*Authenticator, error) {
+	if password == "" {
```

Comment:
It seems inconsistent that `New` and `WithPassword` has different interpretation of the same password parameter. Can we make it that empty password = nil password hash here as well?